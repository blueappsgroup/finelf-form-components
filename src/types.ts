import { ReactElement, ReactNode } from 'react'
import { FieldProps } from 'formik'

// eslint-disable-next-line @typescript-eslint/no-explicit-any
export type FormValuesType = any

export type CheckboxFieldProps = {
  id?: string
  label?: string
  name: string
  required?: boolean
  onChange?: Function
  children?: ReactElement | ReactElement[]
  HTMLcontent?: string
  skipFieldForApi?: boolean
  disableCollapse?: boolean
}

export type FieldWrapProps = FieldProps & {
  label?: string
  showError?: boolean
  onChange?: Function
  placeholder?: string
  required?: boolean
  children?: ReactElement | ReactElement[]
  childrenBody?: ReactElement | ReactElement[]
  type?: string
  suffix?: string
  prefix?: string
  HTMLcontent?: string
  skipFieldForApi?: boolean
  icon?: object
  disableCollapse?: boolean
  visible?: boolean
}

export type FieldDateWrapProps = FieldProps & {
  label?: string
  showError?: boolean
  onChange?: Function
  placeholder?: string
  required?: boolean
  children?: ReactElement | ReactElement[]
  childrenBody?: ReactElement | ReactElement[]
  type?: string
  suffix?: string
  prefix?: string
  dateFormat?: string
}

export type SelectFieldOptions = {
  [key: string]: string
}

export type SelectFieldWrapProps = FieldProps & {
  label?: string
  showError?: boolean
  onChange?: Function
  placeholder?: string
  required?: boolean
  options: SelectFieldOptions
}

export type RangeFieldWrapProps = FieldProps & {
  label?: string
  showError?: boolean
  showMinMax?: boolean
  onChange?: Function
  placeholder?: string
  required?: boolean
  type?: string
  min?: number
  max?: number
  step?: number
  value?: number
  suffix?: string
  unit?: string
}

export type SelectFieldProps = {
  id?: string
  label?: string
  name?: string
  errorMsg?: string
  required?: boolean
  validate?: Function
  showError?: boolean
}

export interface TextFieldProps {
  id?: string
  label?: string
  placeholder?: string
  name: string
  required?: boolean
  validate?: Function
  showError?: boolean
}

export interface TextFieldWithConditonProps extends TextFieldProps {
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  requiredCondition?: { [key: string]: string[] }
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  visibleCondition?: { [key: string]: any }
}

export type RangeFieldProps = {
  id?: string
  label?: string
  name?: string
  errorMsg?: string
  required?: boolean
  showError?: boolean
  showMinMax?: boolean
  min?: number
  max?: number
  step?: number
  value?: number
  suffix?: string
  unit?: string
}

export type EmailFieldProps = {
  id?: string
  label?: string
  placeholder?: string
  name?: string
  errorMsg?: string
  required?: boolean
  showError?: boolean
  icon?: object
}

export type AmountOfChildrenFieldProps = {
  id?: string
  label?: string
  placeholder?: string
  name?: string
  errorMsg?: string
  required?: boolean
  showError?: boolean
}

export type IncomeFieldProps = {
  id?: string
  label?: string
  placeholder?: string
  name?: string
  errorMsg?: string
  required?: boolean
  showError?: boolean
  suffix?: string
}

export type BirthDateFieldProps = {
  id?: string
  label?: string
  placeholder?: string
  name?: string
  errorMsg?: string
  required?: boolean
  showError?: boolean
  dateFormat?: string
}

export type TelephoneFieldProps = {
  id?: string
  label?: string
  placeholder?: string
  name?: string
  errorMsg?: string
  required?: boolean
  showError?: boolean
  validate?: Function
}

export type OtherLoanAmountFieldProps = {
  id?: string
  label?: string
  placeholder?: string
  name?: string
  errorMsg?: string
  required?: boolean
  showError?: boolean
  validate?: Function
}

export type CustomThemeType = {
  fontFamily?: string
  formBoxShadow?: string
  formMaxWidth?: string
  formPaddingMobile?: string
  formPadding?: string
  formBgColor?: string
  styledContainerMobilePadding?: string
  styledContainerPadding?: string
  inputHeight?: string
  inputTextColor?: string
  inputPlaceHolderColor?: string
  inputErrorColor?: string
  inputErrorTextAlign?: string
  inputErrorFontSize?: string
  inputErrorBottom?: string
  inputBorderColor?: string
  inputBorderWidth?: string
  inputBorderStyle?: string
  inputBorderRadius?: string
  inputBgColor?: string
  inputBoxShadow?: string
  inputFontSize?: string
  inputFontWeight?: string
  inputFontStyle?: string
  inputLineHeight?: string
  inputPadding?: string
  styledRowFlexWrap?: string
  styledRowLabelPadding?: string
  styledSelectHeight?: string
  styledSelectTextColor?: string
  styledSelectPlaceHolderColor?: string
  styledSelectErrorColor?: string
  styledSelectErrorTextAlign?: string
  styledSelectBorderColor?: string
  styledSelectBorderWidth?: string
  styledSelectBorderStyle?: string
  styledSelectBorderRadius?: string
  styledSelectBgColor?: string
  styledSelectBoxShadow?: string
  styledSelectFontSize?: string
  styledSelectFontWeight?: string
  styledSelectFontStyle?: string
  styledSelectLineHeight?: string
  styledSelectVerticalPadding?: string
  styledSelectHorizontalPadding?: string
  styledSpanFontSize?: string
  styledSpanFontWeight?: string
  styledRowMarginBottom?: string
  checkboxLabelArrowColor?: string
  checkboxBgColor?: string
  checkboxCheckedBgColor?: string
  checkboxBorderColor?: string
  checkboxBorderErrorColor?: string
  checkboxLabelTextColor?: string
  checkboxMarkColor?: string
  headerTextColor?: string
  headerBorderColor?: string
  subheaderBorderColor?: string
  buttonTextColor?: string
  buttonBgColor?: string
  buttonBorderColor?: string
  successTextColor?: string
  stepHeaderActiveColor?: string
  stepHeaderInActiveColor?: string
  stepHeaderActiveBorderColor?: string
  stepHeaderInActiveBorderColor?: string
  buttonSecondaryTextColor?: string
  buttonSecondaryBgColor?: string
  buttonSecondaryBorderColor?: string
  redirectPageHeaderTextColor?: string
  redirectPageRedirectTextColor?: string
  redirectPageCounterTextColor?: string
  successStatusTextColor?: string
  errorStatusTextColor?: string
  sliderHeight?: string
  sliderBorderRadius?: string
  sliderBgColor?: string
  sliderFillBgColor?: string
  sliderHandleBgColor?: string
  sliderHandleBorderColor?: string
  sliderHandleBoxShadow?: string
  sliderHandleBorderWidth?: string
  sliderHandleBorderStyle?: string
  sliderHandleBorderRadius?: string
  sliderHandleWidth?: string
  sliderHandleHeight?: string
  sliderHandleAfterWidth?: string
  sliderHandleAfterHeight?: string
  sliderHandleAfterBorderRadius?: string
  styledInputSuffixBgColor?: string
  styledInputSuffixTextColor?: string
  styledInputSuffixFontSize?: string
  styledInputSuffixFontWeight?: string
  styledInputSuffixBorderRadius?: string
  styledInputSuffixVerticalPadding?: string
  styledInputSuffixHorizontalPadding?: string
  styledInputPrefixPadding?: string
}

export type HeaderProps = {
  text: string
  type?: 'main' | 'subheader'
  withBorder?: boolean
}

export type ButtonProps = {
  text: string
  variant?: 'primary' | 'secondary'
  type?: 'submit' | 'button'
  size?: 'large | medium | small'
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  onClick?: any
  disabled?: boolean
}

export type FormProps = {
  onSubmit: (
    values: FormValuesType,
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    props?: any
  ) => void
  children: ReactNode
  customTheme?: CustomThemeType
  id: string
  stepsLength?: number
  stepsTitles?: string[]
  hasRedirect?: boolean
  redirectUrl?: string
  timeToRedirect?: number
  redirectHeaderText?: string
  logoImg?: string
  redirectMainImg?: string
  redirectBgImg?: string
  queueUrl?: string
  sendDataToSQS?: boolean
  sendDataToApi?: boolean
  apiUrl?: string
  transactionName?: string
  propertyNamesFromUrl?: string[]
  dataWithUserAgent?: boolean
}

export type RedirectPagePropsTypes = {
  headerText?: string
  backgroundImage?: string
  mainImg?: string
  logoImg?: string
  timeToRedirect?: number
  redirectUrl?: string
}

export type ImageProps = {
  src: string
  alt: string
  text?: string
  hideOnMobile?: boolean
  disableShadow?: boolean
}
