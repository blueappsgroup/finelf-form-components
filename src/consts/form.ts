export const formStatuses = {
  submited: 'submited',
  error: 'error',
}
