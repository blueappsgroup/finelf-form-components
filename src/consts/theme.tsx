import React, { FC, ReactElement } from 'react'
import { ThemeProvider as MainThemeProvider } from 'styled-components'
import { CustomThemeType } from '../types'

export const colors = {
  mainGrey: 'rgba(0, 0, 0, 0.7)',
  ligthGrey: '#ecebed',
  grey: '#B9BABD',
  mediumGrey: '#7C7878',
  darkGrey: '#0f0f0f',
  errorRed: '#FF0000',
  lightRed: '#e2070e',
  mediumRed: '#f20f0f',
  white: '#FFF',
  black: '#000',
}

export const theme = {
  formBgColor: colors.white,
  inputHeight: '63px',
  inputTextColor: colors.mainGrey,
  inputPlaceHolderColor: colors.mainGrey,
  inputErrorColor: colors.errorRed,
  inputErrorTextAlign: 'left',
  inputErrorFontSize: '10px',
  inputErrorBottom: '-12px',
  inputBorderColor: colors.ligthGrey,
  inputBorderWidth: '1px',
  inputBorderStyle: 'solid',
  inputBorderRadius: '6px',
  inputBgColor: colors.white,
  inputBoxShadow: '0px 7px 64px rgba(194, 186, 186, 0.07)',
  inputFontSize: '14px',
  inputFontWeight: '500',
  inputFontStyle: 'normal',
  inputLineHeight: '24px',
  inputTextAlign: 'left',
  inputPadding: '0 20px',
  labelFontSize: '16px',
  datePickerHeight: '63px',
  datePickerTextColor: colors.mainGrey,
  datePickerPlaceHolderColor: colors.mainGrey,
  datePickerBorderColor: colors.ligthGrey,
  datePickerBorderWidth: '1px',
  datePickerBorderStyle: 'solid',
  datePickerBorderRadius: '6px',
  datePickerBgColor: colors.white,
  datePickerBoxShadow: '0px 7px 64px rgba(194, 186, 186, 0.07)',
  datePickerFontSize: '14px',
  datePickerFontWeight: '500',
  datePickerFontStyle: 'normal',
  datePickerLineHeight: '24px',
  datePickerPadding: '0 20px',
  styledRowFlexWrap: 'no-wrap',
  styledRowLabelPadding: '10px',
  styledRowLabelFontColor: colors.black,
  styledSelectHeight: '63px',
  styledSelectTextColor: colors.mainGrey,
  styledSelectPlaceHolderColor: colors.mainGrey,
  styledSelectErrorColor: colors.errorRed,
  styledSelectErrorTextAlign: 'left',
  styledSelectBorderColor: colors.ligthGrey,
  styledSelectBorderWidth: '1px',
  styledSelectBorderStyle: 'solid',
  styledSelectBorderRadius: '6px',
  styledSelectBgColor: colors.white,
  styledSelectBoxShadow: '0px 7px 64px rgba(194, 186, 186, 0.07)',
  styledSelectFontSize: '14px',
  styledSelectFontWeight: '500',
  styledSelectFontStyle: 'normal',
  styledSelectLineHeight: '24px',
  styledSelectVerticalPadding: '0px',
  styledSelectHorizontalPadding: '20px',
  styledSpanFontSize: '16px',
  styledSpanFontColor: colors.black,
  styledSpanFontWeight: 'bold',
  styledRowMarginBottom: '12px',
  checkboxLabelArrowColor: colors.mediumGrey,
  checkboxBgColor: colors.white,
  checkboxBorderColor: colors.black,
  checkboxBorderWidth: '1px',
  checkboxBorderErrorColor: colors.errorRed,
  checkboxLabelTextColor: colors.darkGrey,
  checkboxMarkColor: colors.lightRed,
  headerTextColor: colors.black,
  headerBorderColor: colors.errorRed,
  headerFontSize: '25px',
  headerMargin: '0px 0px 20px 0px',
  subheaderBorderColor: colors.black,
  subheaderFontSize: '16px',
  subheaderMargin: '10px 0px 20px 0px',
  buttonTextColor: colors.white,
  buttonBgColor: colors.mediumRed,
  buttonBorderColor: colors.mediumRed,
  successTextColor: colors.black,
  stepHeaderActiveColor: colors.mediumRed,
  stepHeaderInActiveColor: colors.grey,
  stepHeaderActiveBorderColor: colors.mediumRed,
  stepHeaderInActiveBorderColor: colors.grey,
  buttonSecondaryTextColor: colors.mediumRed,
  buttonSecondaryBgColor: colors.white,
  buttonSecondaryBorderColor: colors.mediumRed,
  redirectPageHeaderTextColor: colors.black,
  redirectPageRedirectTextColor: colors.black,
  redirectPageCounterTextColor: colors.mediumRed,
  successStatusTextColor: colors.black,
  errorStatusTextColor: colors.errorRed,
  sliderHeight: '12px',
  sliderInputHeight: '63px',
  sliderInputTextAlign: 'left',
  sliderBorderRadius: '10px',
  sliderBgColor: colors.mainGrey,
  sliderFillBgColor: colors.mediumRed,
  sliderHandleBgColor: colors.white,
  sliderHandleBorderColor: colors.mainGrey,
  sliderHandleBoxShadow: '0px 7px 64px rgba(194, 186, 186, 0.07)',
  sliderHandleBorderWidth: '1px',
  sliderHandleBorderStyle: 'solid',
  sliderHandleBorderRadius: '50%',
  sliderHandleWidth: '24px',
  sliderHandleHeight: '24px',
  sliderHandleAfterWidth: '7px',
  sliderHandleAfterHeight: '7px',
  sliderHandleAfterBorderRadius: '50%',
  styledInputSuffixBgColor: colors.mediumRed,
  styledInputSuffixTextColor: colors.white,
  styledInputSuffixFontSize: '14px',
  styledInputSuffixFontWeight: 'bold',
  styledInputSuffixBorderRadius: '0 5px 5px 0',
  styledInputPrefixPadding: '0 15px',
}

type ThemeProviderPops = {
  children: ReactElement | ReactElement[]
  customTheme?: CustomThemeType
}

export const ThemeProvider: FC<ThemeProviderPops> = ({
  children,
  customTheme,
}) => (
  <MainThemeProvider theme={{ ...theme, ...customTheme }}>
    {children}
  </MainThemeProvider>
)
