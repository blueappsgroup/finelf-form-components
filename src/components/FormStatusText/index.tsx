import React, { useContext } from 'react'
import { useFormikContext } from 'formik'
import styled from 'styled-components'
import { formStatuses } from '../../consts/form'
import { FormContext } from '../../utils'

type StyledProps = {
  theme: {
    successTextColor: string
    errorStatusTextColor: string
    successStatusTextColor: string
  }
  show: boolean
  isError: boolean
}

const StyledText = styled.span<{ show: boolean; isError: boolean }>`
  display: ${(props: StyledProps): string => (props.show ? 'block' : 'none')};
  color: ${(props: StyledProps): string =>
    props.isError
      ? props.theme.errorStatusTextColor
      : props.theme.successStatusTextColor};
  font-style: normal;
  font-weight: 600;
  font-size: 16px;
  line-height: 18px;
  margin: 10px 0px 20px 0px;
`
type Props = {
  successText: string
  errorText: string
}

const FormStatusText: React.FC<Props> = ({ successText, errorText }) => {
  const { errorFromApi } = useContext(FormContext)
  const { status, dirty } = useFormikContext()
  const isError = status === formStatuses.error || errorFromApi === true
  const showStatusText = (status === formStatuses.submited && !dirty) || isError
  return (
    <StyledText show={showStatusText} isError={isError}>
      {isError ? errorText : successText}
    </StyledText>
  )
}

export default FormStatusText
