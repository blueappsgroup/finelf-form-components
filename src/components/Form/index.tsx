import React, { FC, ReactElement, useMemo, useState } from 'react'
import { Formik, Form } from 'formik'
import styled from 'styled-components'

import { ThemeProvider } from '../../consts/theme'
import { device } from '../../consts/sizes'
import { FormProps, FormValuesType } from '../../types'
import RedirectPage from '../RedirectPage'
import {
  FormContext,
  getFormValuesFromCache,
  resetFormValueCache,
  handleSendDataToApi,
  getFieldsValuesFromUrl,
} from '../../utils'
import { formStatuses } from '../../consts/form'

const StyledForm = styled(Form)`
  display: flex;
  flex-direction: column;
  max-width: ${(props): string => props.theme.formMaxWidth};
  justify-self: center;
  background: ${(props): string => props.theme.formBgColor};
  font-family: ${(props): string => props.theme.fontFamily};
  border-radius: 6px;
  box-shadow: ${(props): string => props.theme.formBoxShadow};
`

const FormWrapper: FC<FormProps> = ({
  children,
  onSubmit,
  customTheme,
  id,
  stepsLength,
  stepsTitles,
  hasRedirect,
  redirectUrl,
  timeToRedirect,
  redirectHeaderText,
  logoImg,
  redirectMainImg,
  redirectBgImg,
  sendDataToApi,
  apiUrl,
  transactionName,
  propertyNamesFromUrl,
  dataWithUserAgent,
}) => {
  const trasationIdValue =
    transactionName &&
    new URLSearchParams(window.location.search).get(transactionName)
  const intialValuesFromUrl = useMemo(
    () =>
      (propertyNamesFromUrl &&
        propertyNamesFromUrl.length > 0 &&
        getFieldsValuesFromUrl(propertyNamesFromUrl)) ||
      {},
    [propertyNamesFromUrl]
  )
  const [redirectUrlPath, setRedirectUrlPath] = useState(redirectUrl)
  const [initialValues, setInitialValues] = useState({
    ...intialValuesFromUrl,
    ...getFormValuesFromCache(id),
    // eslint-disable-next-line @typescript-eslint/camelcase
    trasaction_id: trasationIdValue,
  })
  const [currentStep, setCurrentStep] = useState(0)
  const [fieldsForSkip, setFieldsForSkip] = useState<string[]>([])
  const [errorFromApi, setErrorFromApi] = useState<boolean>(false)
  const shouldRedirect = !errorFromApi && hasRedirect
  const addFieldForSkip = (key: string): void =>
    setFieldsForSkip([...fieldsForSkip, key])

  const handleSubmit = async (
    values: FormValuesType,
    props: {
      resetForm: () => void
      setStatus: (status: string) => void
    }
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
  ): Promise<any> => {
    if (onSubmit) {
      onSubmit(values, props)
    }

    try {
      if (sendDataToApi && apiUrl) {
        const response = await handleSendDataToApi(
          values,
          apiUrl,
          id,
          fieldsForSkip,
          dataWithUserAgent
        )
        const {
          redirectUrl: urlFromApi,
          status: statusFromApi,
        } = await response.json()

        if (statusFromApi === false) {
          setErrorFromApi(true)
          return
        }

        urlFromApi && setRedirectUrlPath(urlFromApi)
      }

      props.resetForm()
      props.setStatus(formStatuses.submited)
    } catch (e) {
      props.setStatus(formStatuses.error)
    }
  }
  const handleReset = (): void => {
    resetFormValueCache(id)
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    setInitialValues({} as any)
  }

  const prevStep: Function = () => setCurrentStep(currentStep - 1)

  const nextStep: Function = () => setCurrentStep(currentStep + 1)

  return (
    <FormContext.Provider
      value={{
        id,
        apiUrl,
        stepsLength,
        currentStep,
        stepsTitleList: stepsTitles,
        nextStep,
        prevStep,
        fieldsForSkip,
        addFieldForSkip,
        errorFromApi,
      }}
    >
      <ThemeProvider customTheme={{ ...customTheme }}>
        <Formik
          enableReinitialize
          initialValues={initialValues}
          onSubmit={handleSubmit}
          onReset={handleReset}
        >
          {(props): ReactElement =>
            (shouldRedirect && props.status === formStatuses.submited && (
              <RedirectPage
                redirectUrl={redirectUrlPath}
                backgroundImage={redirectBgImg}
                logoImg={logoImg}
                headerText={redirectHeaderText}
                timeToRedirect={timeToRedirect}
                mainImg={redirectMainImg}
              />
            )) || (
              <StyledForm id={id}>
                {children}
              </StyledForm>
            )
          }
        </Formik>
      </ThemeProvider>
    </FormContext.Provider>
  )
}

export default FormWrapper
