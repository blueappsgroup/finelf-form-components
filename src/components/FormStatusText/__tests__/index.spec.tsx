import * as React from 'react'
import { render, fireEvent, act } from '@testing-library/react'
import { formStatuses } from '../../../consts/form'
import { FormValuesType } from '../../../types'

import Form from '../../Form'
import FormStatusText from '../index'
describe('<BirthDateField />', () => {
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  const setupWrapper: (config: any) => any = ({
    formId = 'testForm',
    onSubmit = jest.fn(),
  }) => {
    const wrapper = render(
      <Form
        id={formId}
        onSubmit={onSubmit}
        apiUrl="http://localhost"
        sendDataToApi
      >
        <FormStatusText
          successText="Dziękujemy, twoje dane zostały przesłane"
          errorText="Coś poszło nie tak spróbuj ponownie"
        />
      </Form>
    )

    const { container } = wrapper
    const form = container.querySelector(`#${formId}`)

    return {
      form,
      ...wrapper,
    }
  }

  it('matches snapshot', async () => {
    const wrapper = setupWrapper({})
    const { form } = wrapper

    await act(async () => {
      fireEvent.submit(form)
    })

    expect(wrapper).toMatchSnapshot()
    expect(
      wrapper.getByText('Dziękujemy, twoje dane zostały przesłane')
    ).toBeTruthy()
  })

  it('matches snapshot error', async () => {
    const onSubmit = (values: FormValuesType, props: any) => {
      props.setStatus(formStatuses.error)
    }
    const wrapper = setupWrapper({ onSubmit })
    const { form } = wrapper

    await act(async () => {
      fireEvent.submit(form)
    })

    expect(wrapper).toMatchSnapshot()
    expect(
      wrapper.getByText('Coś poszło nie tak spróbuj ponownie')
    ).toBeTruthy()
  })
})
