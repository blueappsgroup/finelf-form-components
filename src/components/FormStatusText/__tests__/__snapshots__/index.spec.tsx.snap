// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BirthDateField /> matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form
        action="#"
        class="sc-iCoHVE JvWob"
        id="testForm"
      >
        <span
          class="sc-fujyUd dkJRbp"
        >
          Dziękujemy, twoje dane zostały przesłane
        </span>
      </form>
    </div>
  </body>,
  "container": <div>
    <form
      action="#"
      class="sc-iCoHVE JvWob"
      id="testForm"
    >
      <span
        class="sc-fujyUd dkJRbp"
      >
        Dziękujemy, twoje dane zostały przesłane
      </span>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "form": <form
    action="#"
    class="sc-iCoHVE JvWob"
    id="testForm"
  >
    <span
      class="sc-fujyUd dkJRbp"
    >
      Dziękujemy, twoje dane zostały przesłane
    </span>
  </form>,
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`<BirthDateField /> matches snapshot error 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form
        action="#"
        class="sc-iCoHVE JvWob"
        id="testForm"
      >
        <span
          class="sc-fujyUd dCNWoi"
        >
          Coś poszło nie tak spróbuj ponownie
        </span>
      </form>
    </div>
  </body>,
  "container": <div>
    <form
      action="#"
      class="sc-iCoHVE JvWob"
      id="testForm"
    >
      <span
        class="sc-fujyUd dCNWoi"
      >
        Coś poszło nie tak spróbuj ponownie
      </span>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "form": <form
    action="#"
    class="sc-iCoHVE JvWob"
    id="testForm"
  >
    <span
      class="sc-fujyUd dCNWoi"
    >
      Coś poszło nie tak spróbuj ponownie
    </span>
  </form>,
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
